include/nn/socket.h

```diff
@@ -6,13 +6,22 @@
 #pragma once
 
 #include <nn/types.h>
+#include <nn/util.h>
 #include <sys/socket.h>
 
-namespace nn {
-namespace socket {
+struct in_addr;
```

Comment:
Is this fwd decl necessary with the sys/socket include?

Reply:
I believe it would be in <netinet/in.h>

Reply:
Yeah you're right, I think a fwd decl is fine then for now